rfcs/0097-no-read-store-dir.md

```diff
@@ -0,0 +1,130 @@
+---
+feature: no-read-store-dir
+start-date: 2021-07-04
+author: Las Safin
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Set the permissions for `/nix/store`, `/nix/store/.links` and `/nix/store/trash` to 1771 instead of 1775, disabling reading the directory for other users.
+
+# Motivation
+[motivation]: #motivation
+
+This is a simple change that acts as an extra layer of security by making
+it harder to access store paths that programs don't need to access.
+
+For a directory, being able to execute without being able to read it means
+that you can access paths inside the directory, if and only if you know the path beforehand,
+since you can not list the directory entries.
+
+In our case, with this change, all users that are not root and are not part of nixbld,
+will only be able to directly access paths inside `/nix/store` which they already know,
+i.e. the hash and the name.
+If you run some program that does not have access to the nix daemon under such
+a user, it will not be able to search the store for built system configurations,
```

Comment:
What exactly do you mean with "search the store" and do applications actually do this? Because I read this as "applications that scrape the store for configuration" and I'm pretty sure I got that wrong.

Reply:
I know of no applications that do this; it is just an example. So yes, applications that scrape the store for configuration, which is what doing `ls /nix/store` is. This is what this RFC is about, scraping the store.

Reply:
Then I suggest removing that line and the following one. There are many legitimate use cases for listing the store (and they'll break and they should be mentioned + alternatives), but "looking for configuration" totally isn't one.

Reply:
That is already in the RFC under Drawbacks.